tests/unit/mathematics/RandomMixin_unittest.cc

```diff
@@ -0,0 +1,176 @@
+#include <shogun/lib/common.h>
+#include <shogun/lib/config.h>
+#include <shogun/base/SGObject.h>
+#include <shogun/mathematics/RandomMixin.h>
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <array>
+
+using namespace shogun;
+
+class MockObject : public CSGObject
+{
+public:
+	const char* get_name() const override
+	{
+		return "MockObject";
+	}
+};
+
+class MockNonRandom : public MockObject
+{
+};
+
+class MockRandom : public RandomMixin<MockObject>
+{
+public:
+	std::mt19937_64& m_prng = RandomMixin<MockObject>::m_prng;
```

Comment:
not sure if i get this one... could you elaborate?

Reply:
m_prng is protected in RandomMixin, so I'm making it public

Reply:
this one i get, but why... i mean a unit test should just test publicly exposed API no need to test internals.... as those changes quite frequently, the important thing that it acts the same way after internal refactor... that way you dont need to rewrite tests

Reply:
I get what you mean. I did this to test reproduciblity when seed is changed, since m_prng is never part of the public interface, however it's part of what defines a random mixin.
would something like this be better?
```
class MockRandom : public RandomMixin<CMockObject>
{
public:
	auto sample() -> decltype(m_prng())
	{
		return m_prng();
	}
};
```